keanu-project/src/main/java/io/improbable/keanu/util/io/MIRLoader.java

```diff
@@ -0,0 +1,69 @@
+package io.improbable.keanu.util.io;
+
+import io.improbable.keanu.network.BayesianNetwork;
+import io.improbable.mir.MIR;
+import io.improbable.mir.SavedBayesNet;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class MIRLoader extends ProtobufLoader {
+
+    @Override
+    public BayesianNetwork loadNetwork(InputStream input) throws IOException {
+        MIR.Model model = MIR.Model.parseFrom(input);
+        checkModelIsCompatible(model);
+
+        return extractGraphAndLoadNetwork(model);
+    }
+
+    private static void checkModelIsCompatible(MIR.Model model) {
+        checkProperties(model);
+        checkEntryPoint(model);
+        checkFunction(model);
```

Comment:
Could check that the model only contains one function? That is implicitly covered by the rule that it can only load Keanu models at the moment, but I'd expect the 'one function' rule to apply for longer than the 'only Keanu models' rule. The 'one function' rule may actually apply indefinitely if function in-lining is carried out by an intermediate tool.

Reply:
We could also check that the one function only contains one graph too. That seems like the set of checks that we'll do to verify that a model can be loaded once we've build the intermediate Bayesnetifier.

Reply:
I've added those checks now.